modules/schematics/src/selector/index.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  Rule,
+  SchematicContext,
+  Tree,
+  apply,
+  applyTemplates,
+  branchAndMerge,
+  chain,
+  filter,
+  mergeWith,
+  move,
+  noop,
+  url,
+} from '@angular-devkit/schematics';
+import {
+  getProjectPath,
+  parseName,
+  stringUtils,
+} from '@ngrx/schematics/schematics-core';
+import { Schema as SelectorOptions } from './schema';
+
+export default function(options: SelectorOptions): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    options.path = getProjectPath(host, options);
+
+    const parsedPath = parseName(options.path, options.name || '');
+    options.name = parsedPath.name;
+    options.path = parsedPath.path;
+
+    const templateSource = apply(url('./files'), [
+      options.spec
+        ? noop()
+        : filter(path => !path.endsWith('.spec.ts.template')),
+      applyTemplates({
+        ...stringUtils,
+        'if-flat': (s: string) =>
+          stringUtils.group(
+            options.flat ? '' : s,
+            options.group ? 'selectors' : ''
+          ),
+        reducerPath: `${relativePath(options)}${stringUtils.dasherize(
+          options.name
+        )}.reducer`,
+        ...(options as object),
+      } as any),
+      move(parsedPath.path),
+    ]);
+
+    return chain([branchAndMerge(chain([mergeWith(templateSource)]))])(
+      host,
+      context
+    );
+  };
+}
+
+function relativePath(options: SelectorOptions) {
+  if (options.feature) {
+    return stringUtils.featurePath(
+      options.group,
+      options.flat,
+      'reducers',
```

Comment:
```suggestion
      'selectors',
```

Reply:
This should stay as `reducers`, we're building up the path to the reducers file here.
The selector spec file will always live next to the selector file so this import will always be `./selector-featureName`.
For reducers we have to navigate to the correct folder because this may vary based on the `group` and `flat` flags.

If we would change this, we would import the reducer as:
```
import * as fromFoo from '../../selectors/foo/foo.reducer'
```